aten/src/ATen/Allocator.h

```diff
@@ -3,14 +3,28 @@
 #include <memory>
 #include <stddef.h>
 
-#include "ATen/Retainable.h"
+#include <ATen/Error.h>
+#include <ATen/Retainable.h>
 
 namespace at {
 
 struct Allocator {
   virtual ~Allocator() {}
-  virtual void* allocate(size_t n) const = 0;
-  virtual void deallocate(void* ptr) const = 0;
+  virtual void* allocate(void* ctx, size_t n) const = 0;
+  virtual void deallocate(void* ctx, void* ptr) const = 0;
```

Comment:
do you think there's any place for making a wrapper struct around the context?  An API with two void *s makes me nervous.

Reply:
Yes, I'm working on this in the next patchset